### PR TITLE
HSTS: support for HTTP Strict Transport Security using libhsts

### PR DESCRIPTION
**Pull request by tangbuulam**
This patch adds two new configure parameters:
 --with-libhsts=PATH to point to libhsts
 --with-hsts-file=FILE to specify location of the dafsa file that
contains the domain database

### CODE REVIEW

**Comment by tangbuulam** [comment]
> This patch adds two new configure parameters:
> --with-libhsts=PATH to point to libhsts
> --with-hsts-file=FILE to specify location of the dafsa file that
> contains the domain database

File: configure.ac [hunk 1]
```diff
@@ -4350,6 +4447,8 @@ AC_MSG_NOTICE([Configured to build curl/libcurl:
   PSL:              ${curl_psl_msg}
   Alt-svc:          ${curl_altsvc_msg}
   HTTP2:            ${curl_h2_msg}
+  HSTS:             ${curl_hsts_msg} (using ${HSTS_FILE})
+  HSTS        :     ${curl_hsts_msg} (using ${HSTS_FILE})
```
Comment (vszakats) [inline configure.ac]: This seems to be a duplicate line here.

File: docs/libcurl/curl_version_info.3 [hunk 2]
```diff
@@ -149,6 +149,9 @@ libcurl was built with support for TLS-SRP. (Added in 7.21.4)
 .IP CURL_VERSION_NTLM_WB
 libcurl was built with support for NTLM delegation to a winbind helper.
 (Added in 7.22.0)
+.IP CURL_VERSION_HSTS
+libcurl was built with support for HSTS
+(Added in 7.62.0)
```
Comment (vszakats) [inline docs/libcurl/curl_version_info.3]: Probably 7.66.0 seems realistic now. Also applies to file `symbols-in-versions`.

**Comment by bagder** [comment]
Duplicate of #2682

**Pull request closed without merging**